src/ape/api/contracts.py

```diff
@@ -234,8 +234,8 @@ def get_handler(abi_type: str) -> Any:
 
 @dataclass
 class ContractContainer:
-    _provider: ProviderAPI
     _contract_type: ContractType
+    _provider: Optional[ProviderAPI]
```

Comment:
Maybe we should add a comment here saying like `"This is only None the user is not connected to a provider."`